pkg/volume/csi/csi_attacher.go

```diff
@@ -592,13 +604,12 @@ func makeDeviceMountPath(plugin *csiPlugin, spec *volume.Spec) (string, error) {
 	if spec == nil {
 		return "", fmt.Errorf("makeDeviceMountPath failed, spec is nil")
 	}
-
-	pvName := spec.PersistentVolume.Name
-	if pvName == "" {
+	specName := spec.Name()
+	if specName == "" {
 		return "", fmt.Errorf("makeDeviceMountPath failed, pv name empty")
 	}
 
-	return path.Join(plugin.host.GetPluginDir(plugin.GetPluginName()), persistentVolumeInGlobalPath, pvName, globalMountInGlobalPath), nil
+	return path.Join(plugin.host.GetPluginDir(plugin.GetPluginName()), persistentVolumeInGlobalPath, specName, globalMountInGlobalPath), nil
```

Comment:
I have added a generated identifier that uses nodeName to avoid collision, PTAL.

Re: FindAttachablePluginByName I have an issue open https://github.com/kubernetes/kubernetes/issues/75494 to follow up on a fix for this to ensure attachability is uniform.  Plus, there's this issue open to enhance the CanAttach check https://github.com/kubernetes/kubernetes/issues/74810.